### PR TITLE
[Stats Refresh] Remove feature flag

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #10975

This removes `statsRefresh` feature flag, associated checks, and all dependencies on `WordPressComStatsiOS`. (Except for `TodayViewController`, which was done in #12767.)

To test:
Basically, verify Stats still works.

- To go Stats. 
  - Verify new Stats appears.
- Set the Today widget.
  - Verify the Today widget uses the correct site.
- Go to 'Blog Posts'.
  - On a post, select 'More' then 'Stats'.
  - Verify 'Post Stats' is displayed.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by danielebogo** [comment]
Hey @ScoutHarris I have a compiler error trying to build this. It's because of `isViewHorizontallyCompact` in `UIViewController+SizeClass` which is part of _WordPressComStatsiOS_.
Do you have the same error?

**Comment by ScoutHarris** [comment]
Hey @danielebogo .

> Do you have the same error?

Sorry about that. I didn't see it until I did clean + build. 

I changed those to use `hasHorizontallyCompactView`, which lives in `WordPressUI:UIViewController+Helpers` and does the same thing.

Ready for another review please.

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+        case .days: return NSLocalizedString("Days", comment: "Title of Days stats filter.")
+        case .weeks: return NSLocalizedString("Weeks", comment: "Title of Weeks stats filter.")
+        case .months: return NSLocalizedString("Months", comment: "Title of Months stats filter.")
+        case .years: return NSLocalizedString("Years", comment: "Title of Years stats filter.")
+        }
+    }
+}
+
+fileprivate extension StatsPeriodType {
+    static let allPeriods = [StatsPeriodType.insights, .days, .weeks, .months, .years]
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: If `StatsPeriodType` conforms to [`CaseIterable`](https://developer.apple.com/documentation/swift/caseiterable) then you can just use `StatsPeriodType.allCases` to get a collection of all its cases

**Comment by ScoutHarris** [comment]
Hey @danielebogo .

The `StatsPeriodType` change is done. Ready for another round please.

**Comment by ScoutHarris** [comment]
Thanks @danielebogo !

**Pull request merged**